src/dbsync/api.rs

```diff
@@ -1992,4 +2013,96 @@ mod tests {
+        assert_eq!(utxo_tokens[0].fingerprint, "asset1gquup8evek9psft57ka29atuv9t4ekujfeulay");
+        assert_eq!(utxo_tokens[0].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[1].id, 86839);
+        assert_eq!(utxo_tokens[1].fingerprint, "asset14hltlww3q8alhqznyz0t5rnz3wrpgg8762lsxk");
+        assert_eq!(utxo_tokens[1].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[2].id, 86784);
+        assert_eq!(utxo_tokens[2].fingerprint, "asset1xh75cc98vzh5s7hw37gzgxuaj67ark2p52lwgv");
+        assert_eq!(utxo_tokens[2].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[3].id, 90128);
+        assert_eq!(utxo_tokens[3].fingerprint, "asset189z77fxt2h758mgth5rc6m2tavsj0s6kepwxf9");
+        assert_eq!(utxo_tokens[3].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[4].id, 86813);
+        assert_eq!(utxo_tokens[4].fingerprint, "asset13nmc52yg556gpu4u53stk8fc5qfhvvyadrd78r");
+        assert_eq!(utxo_tokens[4].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[5].id, 86365);
+        assert_eq!(utxo_tokens[5].fingerprint, "asset10n9gjx0h40un2tl5wpx5dr4h0zg8yqkdc7u468");
+        assert_eq!(utxo_tokens[5].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[6].id, 86923);
+        assert_eq!(utxo_tokens[6].fingerprint, "asset1lg82x7zq3c3fy7w29crkwmnfuv2hj03p2ygmc6");
+        assert_eq!(utxo_tokens[6].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[7].id, 86963);
+        assert_eq!(utxo_tokens[7].fingerprint, "asset16x02ett6wtfjkf4ayz2zfjgec2m6pyr6r6xrmr");
+        assert_eq!(utxo_tokens[7].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[8].id, 100979);
+        assert_eq!(utxo_tokens[8].fingerprint, "asset14e7yfrgurya54e9k83axv3hw3s0kxluz289juu");
+        assert_eq!(utxo_tokens[8].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[9].id, 88257);
+        assert_eq!(utxo_tokens[9].fingerprint, "asset1g5ekxpjwnxu43qaejp3ml8hdfqa38ug7w4z6pw");
+        assert_eq!(utxo_tokens[9].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[10].id, 86920);
+        assert_eq!(utxo_tokens[10].fingerprint, "asset13a25l8trzvf3g2pa4lfpmzdtas6wyrn07rjr9q");
+        assert_eq!(utxo_tokens[10].quantity, BigDecimal::from(1));
+        assert_eq!(utxo_tokens[11].id, 6425);
+        assert_eq!(utxo_tokens[11].fingerprint, "asset1qpcwhg7cvg7wr4m5xa3vd2s79lutkf044pmg4z");
+        assert_eq!(utxo_tokens[11].quantity, BigDecimal::from(199217790));
+    }
+
+    #[tokio::test]
+    #[allow(non_snake_case)]
+    async fn get_utxo_tokens_for_utxo_view_CMW_81() {
+        let dp = crate::DataProvider::new(crate::DBSyncProvider::new(crate::Config {
+            db_path: dotenv::var("DBSYNC_DB_URL").unwrap(),
+        }));
+
+        let utxo_tokens = super::get_utxo_tokens_for_utxo_view(
+            dp.provider(), 
+            3312750, 
+            0
+        ).unwrap();
+
+        assert_eq!(utxo_tokens.len(), 0);
```

Comment:
Are you sure it's useless? Doesn't seem so to me. I think in this case it might be useful because the transaction used does have multiassets, and the length test verifies that those multiassets are excluded in `get_utxo_tokens`, which is a key distinguishing feature from `get_stxo_tokens`.

Reply:
Yes I am sure. 
Unit tests are for testing a single function and not the relation between two different functions.
The test is wrong, at least like it is written right now. The purpose of this function is also to retrieve MultiAssets for a particular UTxO, the difference is that the UTxO is not spent. 

checking for a spent UTxO will give 0 MultiAssets that is true, but it also means the function was not used correctly and actually we maybe should produced a specific error for this case, as this is not obvious for a user. 

The test in any case should check for a unspent TxO which gives proper results. The test like this is not giving an information which validates the function behaves correctly. It basically assumes correct behavior but just makes a partial test, the test for an already spent UTxO which gives a not unique result. We are not even sure if this is actually a UTxO we queried for or just wrong input data which also will gives us no result (an empty list).

Having an empty list as the only verifying information is in most cases not helpful.

Reply:
I'm not sure if it's feasible to unit-test unspent TxO:s. Let's suppose that we test for an unspent TxO. The unit test (in this hypothetical scenario) asserts that a multiasset should be retrieved/displayed and must have certain values in this multiasset, and the unit test passes today. But then a week later this UTxO gets spent in a transaction (on the testnet) and then the unit test starts failing. A unit test like this would require frequent maintenance and is not something we can just put there and leave it. 

I have an idea! I will create a UTxO that is guaranteed to never be used (through a wallet specifically dedicated to this task and then I will "lose" the credentials to that wallet). And then I will put that UTxO in this unit test.

Reply:
Haha yes, maybe as a hint there is also a smart contract which is like a "black hole" everything you sent to it will remain there forever. 
But it is also fine like this, yes you are right the test needs a permanent update as a UTxO can get spent anytime.
The correct way would be to implement the error for this correctly: the UTxO was spent use history endpoint or something like this for a negative test case. Than we could cover it with a test case and on the other side we would need a UTxO from the "always fails" smart contract to have a certain positive test case.